### PR TITLE
Add skeleton project for task-sdk

### LINKED ISSUE #42856: Create `airflow-sdk` dist containing DAG definition level code
This becomes a top-level folder (namespace package) within Airflow package -- including pyproject.toml for building a package and tests

### PR DESCRIPTION
**Pull request by kaxil**
closes https://github.com/apache/airflow/issues/42856 | Part of https://github.com/orgs/apache/projects/405

This PR adds a skeleton project for [AIP-72 Task SDK](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-72+Task+Execution+Interface+aka+Task+SDK)

It creates this new project, integrates them to breeze & runs it on CI.

The code & tests are just placeholders at this point, we will start adding things needs for DAG Authors in separate PRs.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -529,6 +529,7 @@ python_files = [
 ]
 testpaths = [
     "tests",
+    "task_sdk/tests",
```
Comment (ashb) [inline pyproject.toml]: Not sure about this one. (I didn't seem to have to do it for provider tests.)

Comment (kaxil) [reply pyproject.toml]: Yeah `ruff` was detecting usage of "assert" in test file and failing: https://docs.astral.sh/ruff/rules/assert/

I was trying to figure it out what to do and then saw that you already ignored S101: https://github.com/apache/airflow/blob/main/providers/pyproject.toml#L57 so I copied it

So, yeah L532 here can be removed.

File: dev/breeze/src/airflow_breeze/commands/release_management_commands.py [hunk 2]
```diff
@@ -549,6 +555,105 @@ def prepare_airflow_packages(
...
+    check_python_version()
+    perform_environment_checks()
+    fix_ownership_using_docker()
+    cleanup_python_generated_files()
+
+    def _build_package_with_hatch(package_format: str):
+        command = [
+            "hatch",
+            "build",
+            "-c",
```
Comment (ashb) [inline dev/breeze/src/airflow_breeze/commands/release_management_commands.py:583]: Consistency is probably better, but in future we should look at `uv build --project task-sdk`

File: task_sdk/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[tool.hatch.build.targets.wheel]
+packages = ["src/airflow"]
+
+[tool.ruff]
+extend = "../pyproject.toml"
+src = ["src"]
+namespace-packages = ["task_sdk/src"]
```
Comment (ashb) [inline task_sdk/pyproject.toml]: Not sure this one is right/needed?

**Review by ashb — APPROVED** [review]

File: dev/breeze/src/airflow_breeze/commands/release_management_commands.py (hunk 2 above)
Comment (kaxil) [reply dev/breeze/src/airflow_breeze/commands/release_management_commands.py:583]: Yup will create a separate PR to switch it to uv everywhere

File: task_sdk/pyproject.toml (hunk 3 above)
Comment (kaxil) [reply task_sdk/pyproject.toml]: Yeah, removed in https://github.com/apache/airflow/pull/42904/commits/ee669a1e1c0366a345b2f4eb790f4a5e9c3b0688

Comment (ashb) [reply task_sdk/pyproject.toml]: Ah k, this is an attempt at making ruff convert relative imports to the right place.

Without this it thinks `sdk` is a top level package, not `airflow.sdk`. We'll need to play a bit here.

**Comment by rawwar** [comment]
@kaxil , This change is causing mypy pre-commit check to fail with following error:

```
task_sdk/tests/conftest.py:26: error: Incompatible types in assignment
(expression has type "tuple[()]", target has type "Union[str, list[str]]") 
[assignment]
        config.inicfg["airflow_deprecations_ignore"] = ()
```
Should the value be an empty string?

Or, I see in other places, Ash has used `# type: ignore[assignment,operator]` comment.

**Comment by potiuk** [comment]
Nice!

**Pull request merged**